plugins/clock/Clock.cc

```diff
@@ -0,0 +1,91 @@
+#include <gz/plugin/Register.hh>
+#include <gz/sim/System.hh>
+#include <gz/sim/Util.hh>
+#include <yarp/os/Bottle.h>
+#include <yarp/os/BufferedPort.h>
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Network.h>
+#include <yarp/os/Port.h>
+
+using namespace gz;
+using namespace sim;
+using namespace systems;
+
+using yarp::os::Bottle;
+using yarp::os::BufferedPort;
+using yarp::os::Log;
+using yarp::os::Network;
+
+namespace gzyarp
+{
+
+class Clock : public System, public ISystemConfigure, public ISystemPostUpdate, public ISystemReset
+{
+public:
+    Clock()
+        : m_portName("/clock")
+        , m_initialized(false)
+    {
+        yDebug() << "Clock plugin Constructor invoked";
+    }
+
+    ~Clock()
+    {
+        yDebug() << "Clock plugin Destructor invoked";
+        m_clockPort.close();
+    }
+
+    virtual void Configure(const Entity& _entity,
+                           const std::shared_ptr<const sdf::Element>& _sdf,
+                           EntityComponentManager& _ecm,
+                           EventManager& /*_eventMgr*/) override
+    {
+        yDebug() << "Clock plugin Configure invoked";
+
+        if (!m_initialized)
+        {
+            m_initialized = true;
+            if (!m_clockPort.open(m_portName))
+            {
+                yError() << "Failed to open port" << m_portName;
+                return;
+            }
+            yInfo() << "Clock plugin initialized";
+        }
+    }
+
+    void PostUpdate(const UpdateInfo& _info, const EntityComponentManager& _ecm) override
+    {
+        if (_info.paused)
+        {
+            // yDebug() << "Simulation pause, skipping clock update";
```

Comment:
```suggestion
            // yDebug() << "Simulation pause, skipping clock update";
```
Similarly, if a message is commented, we can just remove it from the code to keep code simple.